web/frontend/src/pages/form/components/utils/useQuestionForm.ts

```diff
@@ -32,31 +32,34 @@ const useQuestionForm = (initState: RankQuestion | SelectQuestion | TextQuestion
         case 'addChoiceRank':
           setState({
             ...state,
-            ChoicesMap: newChoicesMap,
+            ChoicesMap: { ChoicesMap: newChoicesMap, URLs: [...ChoicesMap.URLs, ''] },
             MaxN: Math.max(
-              ChoicesMap.get('en').length + 1,
-              ChoicesMap.get('fr').length + 1,
-              ChoicesMap.get('de').length + 1
+              ChoicesMap.ChoicesMap.get('en').length + 1,
+              ChoicesMap.ChoicesMap.get('fr').length + 1,
+              ChoicesMap.ChoicesMap.get('de').length + 1
             ),
             MinN: Math.min(
-              ChoicesMap.get('en').length + 1,
-              ChoicesMap.get('fr').length + 1,
-              ChoicesMap.get('de').length + 1
+              ChoicesMap.ChoicesMap.get('en').length + 1,
+              ChoicesMap.ChoicesMap.get('fr').length + 1,
+              ChoicesMap.ChoicesMap.get('de').length + 1
             ),
           });
           break;
         case 'deleteChoiceRank':
           lang.forEach((lg) => {
-            const filteredChoicesMap = ChoicesMap.get(lg).filter(
+            const filteredChoicesMap = ChoicesMap.ChoicesMap.get(lg).filter(
               (item: string, idx: number) => idx !== optionnalValues
             );
-            ChoicesMap.set(lg, filteredChoicesMap);
+            ChoicesMap.ChoicesMap.set(lg, filteredChoicesMap);
           });
+          ChoicesMap.URLs = ChoicesMap.URLs.filter(
+            (item: string, idx: number) => idx !== optionnalValues
+          );
           setState({
             ...state,
             ChoicesMap: ChoicesMap,
-            MaxN: ChoicesMap.get('en').length,
-            MinN: ChoicesMap.get('en').length,
+            MaxN: ChoicesMap.ChoicesMap.get('en').length,
```

Comment:
Too lazy to follow all the code: why do you default to `en` here instead of using `Math.max`?

Reply:
I just adapted the existing code (also applies to some of your other comments), didn't feel like rewriting too much at this stage as it would imply even more testing... I guess when I add the automated tests for the form creation I'll come back to this code anyway